server/src/main/java/org/elasticsearch/health/HealthIndicatorService.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import java.util.List;
+
+/**
+ * This is a service interface used to report health indicators from the different plugins.
+ */
+public interface HealthIndicatorService {
+
+    List<HealthIndicator> getIndicators();
```

Comment:
I think this calculates the health and should be named "calculate" instead?

I also wonder if we would want to split it more such that each health indicator has it's own service that can calculate the result for that indicator?

So we would have a `HealthIndicatorService` that can deliver a single `HealthIndicatorResult` for a specific named health indicator.